roles/ceph-osd/tasks/pre_requisite.yml

```diff
@@ -58,3 +58,23 @@
   when:
     - cephx
     - item.copy_key|bool
+
+- set_fact:
+     osd_uuids: |
+      {%- set o=[] %}
+      {%- for device in  devices %}
+        {%- if o.append( ( inventory_hostname + device + ansible_date_time.iso8601_micro + "osd_uuids") | to_uuid ) %}
```

Comment:
what i know is if we ceph-disk prepare without --uuid, it will generate random uuid and use this uuid to register and the followed sgdisk command will use the uuid like
```
sgdisk --new=2:0:+10240M --change-name=2:"ceph journal"  --partition-guid=2:ceph-disk-generated-ceph_journal_uuids --typecode=2:45b0969e-9b03-4f30-b4c6-b4b80ceff106 --mbrtogpt --  /dev/sd*

sgdisk --largest-new=1 --change-name=1:"ceph data" --partition-guid=1:ceph-disk-generated-ceph_osd_uuids --typecode=1:4fbd7e29-9d25-41b8-afd0-062c0ceff05d --mbrtogpt -- /dev/sd*
```
maybe
```
inventory_hostname + device  + "osd_uuids") | to_uuid  
```
is better than
```
inventory_hostname + device + ansible_date_time.iso8601_micro + "osd_uuids") | to_uuid
```
and if the device is not formated, this will be no uuid for this devices under /dev/disk/by-uuid/